corehq/apps/app_manager/static/app_manager/js/releases/app_view_release_manager.js

```diff
@@ -43,6 +43,62 @@ hqDefine("app_manager/js/releases/app_view_release_manager", function () {
         $profilesTab.koApplyBindings(profileManager);
     }
 
+    var appReleaseLogsModel = function () {
+        let self = {};
+        self.releaseLogs = ko.observableArray();
+        self.fetchLimit = ko.observable();
+        self.totalItems = ko.observable();
+        self.fetchState = ko.observable();
+
+        self.onPaginationLoad = function () {
+            self.goToPage(1);
+        };
+
+        self.goToPage = function (page) {
+            if (self.fetchState() === 'pending') {
+                return false;
+            }
+            self.fetchState('pending');
+            $.ajax({
+                url: self.getApiEndpoint("paginate_release_logs"),
+                dataType: 'json',
+                data: {
+                    page: page,
+                    limit: self.fetchLimit,
+                },
+                success: function (data) {
+                    self.releaseLogs(
+                        _.map(data.app_release_logs, function (log) {
+                            return ko.mapping.fromJS(log);
```

Comment:
Interesting. What does `ko.mapping.fromJS` do?

Reply:
`ko.mapping.fromJS` generated knockout viewmodels from javascript objects. That makes it easier, and eliminates the need to make the viewmodels ourselves.